model/compile/QtiAssetCompiler/QtiItemAssetXmlReplacer.php

```diff
@@ -47,6 +48,30 @@ public function replaceAssetNodeValue(DOMDocument $domDocument, array $packedAss
             }
         }
 
+        $this->replaceEncodedNodes($xpath, $packedAssets);
+
         return $packedAssets;
     }
+
+    /**
+     * Replace assets in html encoded properties (such as content of text reader interaction)
+     * @param $xpath
+     * @param array $packedAssets
+     */
+    private function replaceEncodedNodes($xpath, array $packedAssets)
+    {
+        $replacementList = array_filter($packedAssets, function ($asset) {
+            return $asset instanceof PackedAsset;
+        });
+        foreach ($replacementList as $key => $asset) {
+            $replacementList[$key] = $asset->getReplacedBy();
+        }
+
+        $attributeNodes = $xpath->query("//*[local-name()='entry']|//*[local-name()='property']") ?: [];
```

Comment:
may i please ask you to put the string inside a constant to have a better overview on what it means 
`//*[local-name()='entry']|//*[local-name()='property']`